app/controllers/notify_controller.rb

```diff
@@ -0,0 +1,62 @@
+class NotifyController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def callback
+    return render_unauthorized if unauthorized?
+    return render_unprocessable_entity if reference_or_status_missing?
```

Comment:
We don't need this if we change the line below to `params.fetch(:reference)` / `params.fetch(:status)`, because that'll cause Rails to do the same thing.